airbyte-cdk/python/unit_tests/sources/file_based/scenarios/csv_scenarios.py

```diff
@@ -3009,6 +3045,61 @@
             ]
         }
     )
-    .set_expected_records([])
     .set_expected_discover_error(AirbyteTracedException, FileBasedSourceError.SCHEMA_INFERENCE_ERROR.value)
 ).build()
+
+csv_no_records_scenario: TestScenario[InMemoryFilesSource] = (
```

Comment:
New scenario - a basic test for if we get 0 records in an expected, okay way. Essentially - if there's no data available, we expect 0 records, and I added this case to make sure that even if 0 records are synced, we get a state message.